app/lib/actions/foreman_ansible/play_hostgroup_roles.rb

```diff
@@ -0,0 +1,65 @@
+module Actions
+  module ForemanAnsible
+    # Actions that initiaztes the playbook run for roles assigned to
+    # the host. It doest that either locally or via a proxy when available.
+    class PlayHostgroupRoles < Actions::EntryAction
+      include ::Actions::Helpers::WithContinuousOutput
+      include ::Actions::Helpers::WithDelegatedAction
+
+      def plan(hostgroup, proxy_selector = ::ForemanAnsible::ProxySelector.new)
+        if hostgroup.hosts.empty?
+          raise ::Foreman::Exception.new(N_('hostgroup is empty'))
+        end
+        input[:hostgroup] = { :id => hostgroup.id, :name => hostgroup.name }
+        proxy = proxy_selector.determine_proxy(hostgroup.hosts[0])
+        inventory_creator = ::ForemanAnsible::InventoryCreator.new(hostgroup.hosts)
+        role_names = []
+        hostgroup.hostgroup_ansible_roles.each do |ansible_role|
+          role_names.append(ansible_role.ansible_role_name)
+        end
+        playbook_creator = ::ForemanAnsible::PlaybookCreator.new(role_names)
```

Comment:
This playbook just contains hostgroup ansible roles, but not all roles for a host. That's unexpected IMO (when I click on 'Play roles' I expect all roles applicable to a host to run, host-specific and hostgroup inherited)